From 2269b039784e2977db62917a3685c7d0490e1588 Mon Sep 17 00:00:00 2001
From: Pascal Quantin <pascal.quantin@gmail.com>
Date: Sat, 25 Aug 2012 21:31:29 +0000
Subject: [PATCH 4/8] From Sylvain Munaut via
 https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=7664
 : packet-gmr1_bcch: Add guards in the SI1/2 choice of
 segment

Although the CSN1 dissector itself will just stop if there is
no matching segment, it will leave the choice field uninitizalized
and so when we use it to fill some other text, it crashes ...

To protect against that, we put a last choice entry that will always
match. As a bonus, it triggers an explicit error in CSN so you
know something is wrong.

svn path=/trunk/; revision=44674
---
 epan/dissectors/packet-gmr1_bcch.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/epan/dissectors/packet-gmr1_bcch.c b/epan/dissectors/packet-gmr1_bcch.c
index f02e149..cbc3d86 100644
--- a/epan/dissectors/packet-gmr1_bcch.c
+++ b/epan/dissectors/packet-gmr1_bcch.c
@@ -884,6 +884,7 @@ CSN_ChoiceElement_t SI1_SegmentChoice[] =
   {7, 0x68, 1, M_TYPE_LABEL(SystemInformation1_t, u.Segment4I,    Segment4I_t,    "Segment 4I")},
   {7, 0x69, 1, M_TYPE_LABEL(SystemInformation1_t, u.Segment4J,    Segment4J_t,    "Segment 4J")},
   {7, 0x6a, 1, M_TYPE_LABEL(SystemInformation1_t, u.Segment4K,    Segment4K_t,    "Segment 4K")},
+  {0, 0x00, 1, CSN_ERROR(SystemInformation1_t, "Unknown segment !", -1)},
 };
 
 static const
@@ -910,6 +911,7 @@ CSN_ChoiceElement_t SI2_SegmentChoice[] =
   {5, 0x04, 1, M_TYPE_LABEL(SystemInformation2_t, u.Segment3E, Segment3E_t, "Segment 3E")},
   {5, 0x06, 1, M_TYPE_LABEL(SystemInformation2_t, u.Segment3G, Segment3G_t, "Segment 3G")},
   {5, 0x0a, 1, M_TYPE_LABEL(SystemInformation2_t, u.Segment3J, Segment3J_t, "Segment 3J")},
+  {0, 0x00, 1, CSN_ERROR(SystemInformation2_t, "Unknown segment !", -1)},
 };
 
 static const
-- 
1.7.10.4

